include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp

```diff
@@ -8,67 +8,70 @@
 #include <future>
 #include <mutex>
 #include <condition_variable>
+#include <chrono>
 
 namespace aikido {
 namespace control {
 
-/// This executor mimics the behavior of BarretFinger. 
-/// It moves a finger to a desired point; it may stop early if 
+/// This executor mimics the behavior of BarretFinger.
+/// It moves a finger to a desired point; it may stop early if
 /// joint limit is reached or collision is detected.
 /// Only the proximal joint is actuated; the distal joint moves with mimic ratio.
 /// When collision is detected on the distal link, the finger stops.
 /// When collision is detected on the proximal link, the distal link moves
 /// until it reaches joint limit or until distal collision is detected.
-class BarrettFingerPositionCommandExecutor
+class BarrettFingerKinematicSimulationPositionCommandExecutor
 {
 public:
   /// Constructor.
-  /// \param _finger Finger to be controlled by this Executor.
-  /// \param _proximal Index of proximal dof
-  /// \param _distal Index of distal dof
-  /// \param _collisionDetector CollisionDetector for detecting self collision
-  ///        and collision with objects. 
-  /// \param _collisionOptions Default is (enableContact=false, binaryCheck=true,
-  ///        maxNumContacts = 1.) 
-  ///        See dart/collison/Option.h for more information 
-  BarrettFingerPositionCommandExecutor(
-    ::dart::dynamics::ChainPtr _finger, int _proximal, int _distal,
-    ::dart::collision::CollisionDetectorPtr _collisionDetector,
-    ::dart::collision::CollisionOption _collisionOptions
+  /// \param[in] finger Finger to be controlled by this Executor.
+  /// \param[in] proximal Index of proximal dof
+  /// \param[in] distal Index of distal dof
+  /// \param[in] collideWith CollisionGroup to check collision with fingers.
+  /// \param[in] collisionOptions Default is (enableContact=false, binaryCheck=true,
+  ///        maxNumContacts = 1.)
+  ///        See dart/collison/Option.h for more information
+  BarrettFingerKinematicSimulationPositionCommandExecutor(
+    ::dart::dynamics::ChainPtr finger, size_t proximal, size_t distal,
+    ::dart::collision::CollisionGroupPtr collideWith,
```

Comment:
The three cases sound good. I would like to add one additional case that `collisionDetector` is `nullptr` and `collideWith` is specified. In this case, we set `mCollisionDetector` to `collideWith->getCollisionDetector()`.

> In particular, I'm unsure what it means for two collision detectors to not be the same. Do they have to be literally the same object or just the same type of collision detector?

They have to be the same _type_, but don't necessarily need to be the same object (but it would be good to be the same object because it could share collision geometries -- this is an implementation detail). The main role of a collision detector is to enforce us to create collision groups of the same collision detection engines (e.g., fcl, bullet, ode). This is necessary because we cannot collide two collision groups created from different collision engines (unless a collision detection engine supports that feature, which I haven't heard about it).

The last bullet point case in my previous comment is that `collideWith`'s collision detector (who created `collideWith`) is not the same type with `collisionDetector`. In this case, we cannot collide HERB hand with `collideWith` because of the reason described above. So we might want to either (1) create a new collision group from the shape frames in `collideWith` through `collisionDetector` or (2) create a collision group of HERB hand through the collision detector type of `collideWith` by `collideWith->getCollisionDetector()->createCollisionGroup(...)`.

Sorry, probably my description was unclear. :sweat: